rust/sdk-processor/src/steps/parquet_default_processor/gcs_handler.rs

```diff
@@ -0,0 +1,185 @@
+}
+
+#[async_trait]
+impl Uploadable for GCSUploader {
+    async fn handle_buffer(
+        &mut self,
+        buffer: ParquetTypeStructs,
+    ) -> anyhow::Result<(), ProcessorError> {
+        // Directly call `upload_generic` for each buffer type
+        let table_name = buffer.get_table_name();
+
+        let result = match buffer {
+            ParquetTypeStructs::Transaction(transactions) => {
+                self.upload_generic(&transactions[..], ParquetTypeEnum::Transaction, table_name)
+                    .await
+            },
+            ParquetTypeStructs::MoveResource(resources) => {
+                self.upload_generic(&resources[..], ParquetTypeEnum::MoveResource, table_name)
+                    .await
+            },
+            ParquetTypeStructs::WriteSetChange(changes) => {
+                self.upload_generic(&changes[..], ParquetTypeEnum::WriteSetChange, table_name)
+                    .await
+            },
+            ParquetTypeStructs::TableItem(items) => {
+                self.upload_generic(&items[..], ParquetTypeEnum::TableItem, table_name)
+                    .await
+            },
+            ParquetTypeStructs::MoveModule(modules) => {
+                self.upload_generic(&modules[..], ParquetTypeEnum::MoveModule, table_name)
+                    .await
+            },
+        };
+
+        if let Err(e) = result {
+            error!("Failed to upload buffer: {}", e);
+            return Err(ProcessorError::ProcessError {
+                message: format!("Failed to upload buffer: {}", e),
+            });
+        }
+        Ok(())
+    }
+}
+
+pub fn create_new_writer(schema: Arc<Type>) -> anyhow::Result<SerializedFileWriter<Vec<u8>>> {
+    let props = WriterProperties::builder()
+        .set_compression(parquet::basic::Compression::LZ4)
+        .build();
+    let props_arc = Arc::new(props);
+
+    SerializedFileWriter::new(Vec::new(), schema, props_arc).context("Failed to create new writer")
+}
+
+impl GCSUploader {
+    pub fn new(
+        gcs_client: Arc<GCSClient>,
+        parquet_type_to_schemas: HashMap<ParquetTypeEnum, Arc<Type>>,
+        parquet_type_to_writer: HashMap<ParquetTypeEnum, SerializedFileWriter<Vec<u8>>>,
+        bucket_name: String,
+        bucket_root: String,
+        processor_name: String,
+    ) -> anyhow::Result<Self> {
+        Ok(Self {
+            gcs_client,
+            parquet_type_to_schemas,
+            parquet_type_to_writer,
+            bucket_name,
+            bucket_root,
+            processor_name,
+        })
+    }
+
+    fn create_new_writer(
+        &self,
+        parquet_type: ParquetTypeEnum,
+    ) -> anyhow::Result<SerializedFileWriter<Vec<u8>>> {
+        let schema = self
+            .parquet_type_to_schemas
+            .get(&parquet_type)
+            .context("Parquet type not found in schemas")?
+            .clone();
+
+        create_new_writer(schema)
+    }
+
+    fn close_writer(
```

Comment:
I don't understand what this function actually does. Could you add some comments please? Why do we need to replace the old writer? Is that so it creates an empty buffer? It also doesn't actually close any writers. Maybe rename to `get_and_replace_writer`?

Reply:
the reason why we replace is once we’re ready to upload (either because the buffer is full or enough time has passed), we don’t want to keep adding new data to that same writer.  we want a clean slate for the next batch.

So, we **replace** the old writer with a new one to empty the writer buffer without losing any data. Otherwise, I remember i was having a weird state where the parquet file has mixed data so that it wasn't properly formatted and couldn't be loaded into BigQuery. 

makes sense to rename it though